configuration/src/main/java/com/synopsys/integration/configuration/config/PropertyConfiguration.java

```diff
@@ -109,6 +132,13 @@ public <V, R> Optional<V> getProvidedParsedValue(@NotNull TypedProperty<V, R> pr
             return value.getValue();
         } else if (parseException.isPresent() && propertyResolutionInfo.isPresent()) {
             throw new InvalidPropertyException(property.getKey(), propertyResolutionInfo.get().getSource(), parseException.get());
+        } else if(!scanSettingsProperties.isEmpty() && scanSettingsProperties.containsKey(property.getKey())) {
```

Comment:
Quite a bit of duplicated code in this file. Is it a good time to deduplicate it?

Reply:
Now is as good as any time to deduplicate.

Reply:
If you see return types for all the methods, they are all different from each other. So I believe it would be very tedious to deduplicate all of this code at this point. Might raise bugs that we do not want.